stdlib/public/stubs/Stubs.cpp

```diff
@@ -511,6 +513,8 @@ void swift::_swift_stdlib_flockfile_stdout() {
 void swift::_swift_stdlib_funlockfile_stdout() {
 #if defined(_WIN32)
   _unlock_file(stdout);
+#elif defined(__wasi__)
+  // WebAssembly/WASI doesn't support file locking yet
```

Comment:
Please create a JIRA issue for supporting threading/locking on WASI and reference it here.